idris package missing gmp.h, can't compile hello world.

## Issue description

Trying to compile fails with 

```
~/projects/idrissandbox 2m 6s
❯ idris -o main Main.idr
In file included from /nix/store/4sbvpzm6lvjr9cz8cmghjhb35i60irhj-idris-1.3.1-data/share/ghc-8.6.3/x86_64-linux-ghc-8.6.3/idris-1.3.1/rts/idris_rts.h:522:0,
                 from /tmp/idris5886-0.c:2:
/nix/store/4sbvpzm6lvjr9cz8cmghjhb35i60irhj-idris-1.3.1-data/share/ghc-8.6.3/x86_64-linux-ghc-8.6.3/idris-1.3.1/rts/idris_gmp.h:5:10: fatal error: gmp.h: No such file or directory
 #include <gmp.h>
          ^~~~~~~
compilation terminated.
FAILURE: "/nix/store/isg8rxaxkijl9x3hr2gzsf8pqfnqxg3k-gcc-wrapper-7.4.0/bin/gcc" ["-fwrapv","-fno-strict-overflow","-std=c99","-pipe","-fdata-sections","-ffunction-sections","-D_POSIX_C_SOURCE=200809L","-DHAS_PTHREAD","-DIDRIS_ENABLE_STATS","-I.","-Wl,-gc-sections","/tmp/idris5886-0.c","-L/nix/store/4sbvpzm6lvjr9cz8cmghjhb35i60irhj-idris-1.3.1-data/share/ghc-8.6.3/x86_64-linux-ghc-8.6.3/idris-1.3.1/rts","-lidris_rts","-lgmp","-DIDRIS_GMP","-lpthread","-I/nix/store/4sbvpzm6lvjr9cz8cmghjhb35i60irhj-idris-1.3.1-data/share/ghc-8.6.3/x86_64-linux-ghc-8.6.3/idris-1.3.1/rts","-lm","-I.","-I/nix/store/mma2ff9v5c7s50r9md8xxfw9qv58bfh3-idris-1.3.1/libs/prelude","-I/nix/store/mma2ff9v5c7s50r9md8xxfw9qv58bfh3-idris-1.3.1/libs/base","-I.","-I/nix/store/wsm1zzcrhlc0scs4hfsrv05ba1932d75-idris-1.3.1/libs/prelude","-I/nix/store/wsm1zzcrhlc0scs4hfsrv05ba1932d75-idris-1.3.1/libs/base","-o","main"]

```

### Steps to reproduce
Install `idris` (not `idrisPackages.idris`)
Create Main.idr with 

```
module Main

main : IO ()
main = putStrLn "Hello, World"
```
run `idris -o main Main.idr`

observe error


## Technical details

```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.20.3, NixOS, 19.03pre167225.1b3affcbaa8 (Koi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2`
 - channels(root): `"nixos-19.03pre167225.1b3affcbaa8"`
 - channels(aria): `"home-manager"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/development/idris-modules/idris-wrapper.nix
+++ b/pkgs/development/idris-modules/idris-wrapper.nix
@@ -1,4 +1,4 @@
-{ lib, symlinkJoin, makeWrapper, idris-no-deps, gcc, gmp }:
+{ stdenv, lib, symlinkJoin, makeWrapper, idris-no-deps, gcc, gmp }:
 
 symlinkJoin {
   inherit (idris-no-deps) name src meta;
@@ -7,6 +7,8 @@ symlinkJoin {
   postBuild = ''
     wrapProgram $out/bin/idris \
       --run 'export IDRIS_CC=''${IDRIS_CC:-${lib.getBin gcc}/bin/gcc}' \
-      --suffix LIBRARY_PATH : ${lib.makeLibraryPath [ gmp ]}
+      --set NIX_CC_WRAPPER_${stdenv.cc.infixSalt}_TARGET_HOST 1 \
+      --prefix NIX_CFLAGS_COMPILE " " "-I${lib.getDev gmp}/include" \
+      --prefix NIX_CFLAGS_LINK " " "-L${lib.getLib gmp}/lib"
   '';
 }

--- a/pkgs/development/idris-modules/idris-wrapper.nix
+++ b/pkgs/development/idris-modules/idris-wrapper.nix
@@ -1,4 +1,4 @@
-{ stdenv, lib, symlinkJoin, makeWrapper, idris-no-deps, gcc, gmp }:
+{ stdenv, lib, symlinkJoin, makeWrapper, idris-no-deps, gmp }:
 
 symlinkJoin {
   inherit (idris-no-deps) name src meta;
@@ -6,7 +6,7 @@ symlinkJoin {
   buildInputs = [ makeWrapper ];
   postBuild = ''
     wrapProgram $out/bin/idris \
-      --run 'export IDRIS_CC=''${IDRIS_CC:-${lib.getBin gcc}/bin/gcc}' \
+      --run 'export IDRIS_CC=''${IDRIS_CC:-${stdenv.cc}/bin/cc}' \
       --set NIX_CC_WRAPPER_${stdenv.cc.infixSalt}_TARGET_HOST 1 \
       --prefix NIX_CFLAGS_COMPILE " " "-I${lib.getDev gmp}/include" \
       --prefix NIX_CFLAGS_LINK " " "-L${lib.getLib gmp}/lib"

